src/framebuffer/core.rs

```diff
@@ -25,20 +26,37 @@ pub struct Framebuffer<'a> {
     /// like it has been done in `Framebuffer::new(..)`.
     pub var_screen_info: VarScreeninfo,
     pub fix_screen_info: FixScreeninfo,
+    pub swtfb_client: Option<super::swtfb_client::SwtfbClient>,
 }
 
 unsafe impl<'a> Send for Framebuffer<'a> {}
 unsafe impl<'a> Sync for Framebuffer<'a> {}
 
 impl<'a> framebuffer::FramebufferBase<'a> for Framebuffer<'a> {
     fn from_path(path_to_device: &str) -> Framebuffer<'_> {
-        let device = OpenOptions::new()
-            .read(true)
-            .write(true)
-            .open(path_to_device)
-            .unwrap();
+        let swtfb_client = if path_to_device == "/dev/shm/swtfb.01" {
```

Comment:
Let's define the string once and use it everywhere. Maybe as a method on the model gen enum?
```rust
impl Model {
    pub framebuffer_path(&self) -> &'static str {
        match *self {
            Model::Gen1 => "/dev/fb0",
            Model::Gen2 => "/dev/shm/swtfb.01",
        }
    }
}
```
thus the device `get_framebuffer_path()` also becomes:
```rust
    pub fn get_framebuffer_path(&self) -> &'static str {
        self.model.framebuffer_path()
    }
```
and here: (this is surely wrong syntax but you get the idea)
```rust
        let swtfb_client = if path_to_device == (Model::Gen2).framebuffer_path() {
```

Reply:
Good point!

Reply:
Should be done with commit 6f270a8.